### ISSUE TITLE
Run autograder when invoked from current module

### ISSUE THREAD

**Issue by sravanareddy** [sravanareddy/cs111grader#27]
Thinking of an argumentless function that can we can put in the skeleton file, like this

```
if __name__=='__main__':
    import autograder
    autograder.checkme()

```

which will run tests only on that question, so that students can run their own code directly. This may involve some non-trivial modification to how we currently load modules.

An easier alternative might be to invoke the entire `autograder.py` as is, using Python's [runpy](https://docs.python.org/2/library/runpy.html) module. 

Observing students yesterday, I realized that constantly switching to `autograder.py` from their code file within Canopy is painful, so this is an important feature. Our current process is really designed to be run from the shell.

**Comment by dkreimer** [comment]
So the autograder.py file actually has a variable questionToGrade and if you define it as <code>'q1'</code> it only grades q1, and etc

**Comment by sravanareddy** [comment]
Right... So I was thinking of running the whole autograder module from
the code with the specific q id. Does it work? The pset code may need
to be in the same directory.

**Comment by dkreimer** [comment]
Yeah, in order to import autograder it needs to be in the same directory right? But if it is then you can wrap the lines under if name=main in a function that has an optional argument being the q id, and then invoke it in other modules

**Comment by sravanareddy** [comment]
Yeah -- can you do that and see how it does? Just remember not to push pset solutions to the main repo for privacy reasons. (You can copy or soft link the files so they're in the same directory.)

If this simple solution doesn't work, we'll look into more complex alternatives.

**Comment by dkreimer** [comment]
It doesn't work - if you call <code>autograder.main(question='q4')</code> from hourglass.py, then in importing the modules, evaluate runs hourglass.py, which calls main, which calls hourglass which calls main etc, so it's an endless recursion.
